RecoHI/HiEvtPlaneAlgos/src/HiEvtPlaneFlatProducer.cc

```diff
@@ -69,36 +82,68 @@ using std::vector;
 //
 
 class HiEvtPlaneFlatProducer : public edm::EDProducer {
-   public:
-      explicit HiEvtPlaneFlatProducer(const edm::ParameterSet&);
-      ~HiEvtPlaneFlatProducer();
-
-   private:
-      virtual void beginJob() override ;
-      virtual void produce(edm::Event&, const edm::EventSetup&) override;
-      virtual void endJob() override ;
-      
-      // ----------member data ---------------------------
-
-  edm::EDGetTokenT<reco::VertexCollection> vtxCollection_;
-  edm::EDGetTokenT<reco::EvtPlaneCollection> inputPlanes_;
-  edm::EDGetTokenT<int> centrality_;
-
-  int vs_sell;   // vertex collection size
-  float vzr_sell;
-  float vzErr_sell;
-
-  Double_t epang[NumEPNames];
+public:
+  explicit HiEvtPlaneFlatProducer(const edm::ParameterSet&);
+  ~HiEvtPlaneFlatProducer();
+  
+private:
+  virtual void beginJob() ;
+  virtual void produce(edm::Event&, const edm::EventSetup&);
+  virtual void endJob() ;
+  
+  // ----------member data ---------------------------
+  
+  
+  std::string centralityVariable_;
+  std::string centralityLabel_;
+  std::string centralityMC_;
+  
+  edm::InputTag centralityBinTag_;
+  edm::EDGetTokenT<int> centralityBinToken;
+  edm::Handle<int> cbin_;
+  
+  edm::InputTag centralityTag_;
+  edm::EDGetTokenT<reco::Centrality> centralityToken;
+  edm::Handle<reco::Centrality> centrality_;
+  
+  edm::InputTag vertexTag_;
+  edm::EDGetTokenT<std::vector<reco::Vertex>> vertexToken;
+  edm::Handle<std::vector<reco::Vertex>> vertex_;
+  
+  
+  edm::InputTag inputPlanesTag_;
+  edm::EDGetTokenT<reco::EvtPlaneCollection> inputPlanesToken;
+  edm::Handle<reco::EvtPlaneCollection> evtPlanes_;
+  
+  edm::InputTag trackTag_;
+  edm::EDGetTokenT<reco::TrackCollection> trackToken;
+  edm::Handle<reco::TrackCollection> trackCollection_;
+  
+  int FlatOrder_;
+  double caloCentRef_;
+  double caloCentRefWidth_;
+  int caloCentRefMinBin_;
+  int caloCentRefMaxBin_;
+  int NumFlatBins_;
+  int CentBinCompression_;
+  int Noffmin_;
+  int Noffmax_;
+  double ntrkval;
+  unsigned int runno_;
+  bool FirstEvent;
   HiEvtPlaneFlatten * flat[NumEPNames];
```

Comment:
it looks like this array doesn't need to be of pointers, just use ```HiEvtPlaneFlatten flat[NumEPNames]; ```
I'd say it shouldn't even be in this producer data members: they are used only in ::produce method.
Move it to ::produce